### PR TITLE
refactor(ux): Misc UX Improvements (tutorial copy, caching, filters)

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

In this PR, we address a few small UX issues.

1. We rephrase various copy in the Onboarding Tutorial 
2. Fix user and group caching to support multi-page loading
3. Auto-expand selected filters 
4. Move Platform filter up higher based on usage evaluation

## Summary

Ready for review. 

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/identity/group/GroupList.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const GroupList = () => {
     };
 
     const handleDelete = (urn: string) => {
-        removeGroupFromListGroupsCache(urn, client);
+        removeGroupFromListGroupsCache(urn, client, page, pageSize);
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/identity/group/GroupList.tsx:69]: nice thanks for the fix. We should check all multi-page things (note to myself too)

File: datahub-web-react/src/app/onboarding/config/EntityProfileOnboardingConfig.tsx [hunk 2]
```diff
@@ -35,8 +35,8 @@ export const EntityProfileOnboardingConfig: OnboardingStep[] = [
         content: (
             <Typography.Paragraph>
                 <p>
-                    You can view and edit an entity&apos;s key-value <strong>Properties</strong> on this tab. These are
-                    sourced from the original Data Platform.
+                    You can view an entity&apos;s key-value <strong>Properties</strong> on this tab. These are sourced
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/onboarding/config/EntityProfileOnboardingConfig.tsx:38]: Appreciate you fixing this!

File: datahub-web-react/src/app/onboarding/config/IngestionOnboardingConfig.tsx [hunk 3]
```diff
@@ -13,9 +13,9 @@ export const IngestionOnboardingConfig: OnboardingStep[] = [
         content: (
             <Typography.Paragraph>
                 <p>
-                    Click here to configure new Integrations from DataHub to your <strong>Data Platforms</strong>,
-                    including Transactional Databases like <strong>MySQL</strong>, Data Warehouses like{' '}
-                    <strong>Snowflake</strong>, Dashboarding tools like <strong>Looker</strong>, and many more!
+                    Configure new Integrations from DataHub to your <strong>Data Platforms</strong>, including
+                    Transactional Databases like <strong>MySQL</strong>, Data Warehouses such as{' '}
+                    <strong>Snowflake</strong>, Dashboarding tools like <strong>Looker</strong>, and more!
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/onboarding/config/IngestionOnboardingConfig.tsx:18]: Wording should be consistent for dashboarding, we should say "Dashboarding tools such as Looker"

File: datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx [hunk 4]
```diff
@@ -15,12 +15,13 @@ export const PoliciesOnboardingConfig: OnboardingStep[] = [
                     Welcome to DataHub <strong>Policies</strong>!
                 </p>
                 <p>
-                    If you need fine-grained access controls, <strong>Policies</strong> will do the trick.
+                    In most cases, <strong>Roles</strong> can be used to grant privileges to users on DataHub.
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx:18]: This seems to imply that Roles are not usable in some cases, which isn't the case. What we're really trying to get across is that sometimes they're not the best fit.

Can we say something like "In most cases, **Roles** are the best option for granting privileges to DataHub users."

File: datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx [hunk 5]
```diff
@@ -15,12 +15,13 @@ export const PoliciesOnboardingConfig: OnboardingStep[] = [
...
                 </p>
                 <p>
-                    For most users, <strong>Roles</strong> are the recommended way to manage permissions on DataHub. If
-                    Roles do not fit your use case, then Policies can be used.
+                    When more fine-grained control over user and group permissions is required, then{' '}
+                    <strong>Policies</strong> will do the trick.
                 </p>
+                <p>Policies allow you to grant fine-grained privileges to specific users and groups.</p>
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx:24]: This seems redundant with the previous line

File: datahub-web-react/src/app/search/SimpleSearchFilters.tsx [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { useEffect, useState } from 'react';
 import { FacetFilterInput, FacetMetadata } from '../../types.generated';
 import { SimpleSearchFilter } from './SimpleSearchFilter';
 
-const TOP_FILTERS = ['degree', 'entity', 'tags', 'glossaryTerms', 'domains', 'owners'];
+const TOP_FILTERS = ['degree', 'entity', 'platform', 'tags', 'glossaryTerms', 'domains', 'owners'];
```
Comment (aditya-radhakrishnan) [inline datahub-web-react/src/app/search/SimpleSearchFilters.tsx:6]: Awesome!

**Review by aditya-radhakrishnan — APPROVED** [review]
Please fix the wording for some of the onboarding steps, but otherwise looks great!

File: datahub-web-react/src/app/onboarding/config/IngestionOnboardingConfig.tsx (hunk 3 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/onboarding/config/IngestionOnboardingConfig.tsx:18]: I disagree. Typically by changing it sounds less repetitive, more natural

File: datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx (hunk 4 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx:18]: Very much agree!

File: datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx (hunk 5 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/onboarding/config/PoliciesOnboardingConfig.tsx:24]: agree

**Pull request merged**